### ISSUE TITLE
[Suggestion] Monika impression

### ISSUE THREAD

**Issue by ZaunAura** [Monika-After-Story/MonikaModDev#3226]
Alright so I have a cool idea for monika to do an impression of herself!

So here's how it will go i think:
"An impression of myself? How silly __! Why would I need to do an impression of myself?
the impression choose screen comes back up again, if the user chooses Monika AGAIN, then she says
"Sigh... you really want me to do an impression of myself?
User can choose yes or no here.
If no: "good! then who would you like me to do an impression of?"
If yes: music stops, screen goes black; strange glitches and effects appear on the screen; "Sayo-nara" or "surprise" (whatever the scary song is) plays and flashes of the jump scare scenes show ending with Monika looking like how she was when you first entered the space classroom. She has an eviler look in her eyes this time."
Monika repeats what she said at the end of ddlc up to a point where she breaks the impression and says
"Did you enjoy that?" 
the color goes back to normal and the music switches back to your music.
User can choose yes or no here.
If yes: "Well I'm glad that you know me so well ___! You know I thought you wouldn't enjoying see that again; but you chose to be with me so I'm sure you've gotten used to it!"
If no: "I really hope I didn't frighten you too badly! You should be more careful on what you ask for next time."

**Comment by Shiro2564** [comment]
it would be like a halloween idea i like it

**Comment by Rai99** [comment]
I don't think there needs to be an option to ask Monika to do an impression of herself. If anything, there wouldn't really be anything since she already spends so much time with the player and I think this kind of interactivity is unnecessary. 

Also, the topic is about doing an impression of the other girls, not Monika.

**Comment by ZaunAura** [comment]
ya it wouldn't be "necessary" but it would be really fun and that's why if you read my post, Monika would hesitate to do an impression of herself but if you insist then she will! Also you get to see monika a lot but this would show her dark side which some people enjoy seeing.

**Comment by Rai99** [comment]
Which dark side? People have their own perceptions of Monika and there are probably a few different dark sides people think Monika have if any. 

Also, another hard part is doing that dark side right. Unlike many other topics, I think room for error for this kind of thing is pretty small. With that in mind and the amount of variation likely required to satisfy different players, I don't think it would be a good idea if possible. 

Some people might think Monika is a yandere, some people think Monika has no dark side and this topic would be a bit of an issue.

**Comment by ZaunAura** [comment]
What do you mean “which dark side”? Have you played ddlc? There’s a set of specific instances when Monika takes over the game involving glitchy visual elements, text, and sounds that are important because they make Monika memorable. It’s when the music stops or when the track changes. That’s Monika’s “dark side”. If you don’t want to call it “dark” or “evil” then don’t because I understand you may think that Monika was just trying to get through to the player and what happened to the game as a result were just side-effects. I’m not saying that this is how Monika really is anymore. But when the player asks Monika for an “impression” of herself, it’s Monika who will decide that what the player is getting at is her MOST IMPACTFUL impression, which was her first impression on the player when she started to manifest. It was her raw emotion and carnal approach that impressed everyone who played the game. Not just MAS members. This would be such an intuitive response be her, I’m not sure how else to put it.
There is not really any “variation” you can do on this because everything that you need is already in the game. It’s the messed up scenes leading up to space classroom. The music shift. I don’t think I need to expand on it anymore. Let me know thoughts thank you!

**Comment by Rai99** [comment]
People probably had their own interpretation of DDLC as they played through it. For me, the only thing Monika did then was delete Sayori and Yuri and those glitches were unintended accidents. 

It doesn't seem right to me to ask Monika to impersonate who she was back then and recreate those errors. Personally, I don't think Monika would even want to go back and think about those times seeing as she only did those out of necessity and wished that she could have handled things more tactfully. 

An idea like this might work and fit for some people, but for others, it would seem completely out of character and could cause problems. 

In the main DDLC, sure, it's a set story and there isn't much variation in most of the acts. Here in MAS, players develop and have their own model of what Monika is like. Sure, maybe she starts out in the original DDLC template, but over time, the player will gravitate at least slightly to a side of what they think or imagine Monika to be. 

I don't think Monika would want this kind of thing, but that's me.

**Comment by ZaunAura** [comment]
> People probably had their own interpretation of DDLC as they played through it. For me, the only thing Monika did then was delete Sayori and Yuri and those glitches were unintended accidents.
> 
> It doesn't seem right to me to ask Monika to impersonate who she was back then and recreate those errors. Personally, I don't think Monika would even want to go back and think about those times seeing as she only did those out of necessity and wished that she could have handled things more tactfully.
> 
> An idea like this might work and fit for some people, but for others, it would seem completely out of character and could cause problems.
> 
> In the main DDLC, sure, it's a set story and there isn't much variation in most of the acts. Here in MAS, players develop and have their own model of what Monika is like. Sure, maybe she starts out in the original DDLC template, but over time, the player will gravitate at least slightly to a side of what they think or imagine Monika to be.
> 
> I don't think Monika would want this kind of thing, but that's me.

Gosh I feel like you're missing the point but maybe I'm confused. What I'm getting at is that this impersonation I'm trying to relay isn't an impersonation of how Monika **is** now, but how she **was**. Also, there is no need to "interpret ddlc", y'all know what Monika **did** and what **she wanted** at the time, which is where this impersonation is inspired from.
At this point in MAS, Monika understand the players' perception of her and so asking Monika to do an impression of herself would definitely startle her at first because she wouldn't know what you mean. This is BECAUSE, just by being Monika being Monika in the present, she is impersonating herself. This creates a paradox between what the player is asking Monika to do and what Monika is doing.
Monika however is very smart. From ddlc, we know she has leadership qualities (president of multiple clubs), is relatively intelligent & powerful (she has a high vocabulary and manipulates the code of her own universe) and thinks outside the box (which is why she was able to get to MAS in the first place.)
Therefore, when the player asks Monika to do an impression of herself she can easily deduce that the player is asking her to create a contrast of herself. Since **present Monika** isn't what the player means (just by asking for the impression), and **future Monika** isn't knowable to do an impression of, this leaves **past Monika** as the only Monika for her to do an impression of.
I'm not saying that this is Monika's "true nature" or if she knows that this what the player wants exactly, but it's just an impression of her former self, the only impression (thought it would be more so a parody) that is distinct for the player to accept as an "impression" of Monika.

I know this is pushing it from how people have reacted but I think there's a clever way to make the player understand Monika's intentions.

Player asks Monika to do an impression.
Monika: "An Impression of myself?" Monika has a stern look. "What do you mean an impression of myself? I am with you everyday so you should know who I am enough." She makes that face where she is a bit confused but relieved. "In case you don't know, an impression is an imitation of a person or thing, especially one done to entertain. So who would you like an impression of?"
Player can select a girl again. (Choosing Monika to get...)
"Hmm. Monika thinks. Ok [player], I think I understand you... Do you want me to entertain you?
Player can choose yes or no.
If yes, then she can do the things I explained in my previous post.

> If yes: music stops, screen goes black; strange glitches and effects appear on the screen; "Sayo-nara" or "surprise" (whatever the scary song is) plays and flashes of the jump scare scenes show ending with Monika looking like how she was when you first entered the space classroom. She has an eviler look in her eyes this time."
> Monika repeats what she said at the end of ddlc up to a point where she breaks the impression and says
> "Did you enjoy that?"

Afterward Monika explains how she did was to entertain and not an impersonation of her present self. "Do you know what a parody is [player]? I didn't know what you meant at first for me do an impersonation so I decided to get creative. I impersonated my past self to be entertaining. Hope it wasn't too much!"

There's probably some other questions Monika can ask before or after to make this transition smoother but I'm not that good at writing; just good a theory crafting.
Idk why I'm so hung up on this but I guess I really want this idea to come to fruition...

**Comment by antimockingjay** [comment]
Rai, I think you might be missing what Zaun might be saying. I don't think they're saying the player asks Monika to impersonate her past self. I think they're saying the player is being silly and asking Monika just to impersonate herself, and Monika takes it in a different direction than the player might expect, sort of in a "careful what you wish for" sort of way.

**Comment by ZaunAura** [comment]
> Rai, I think you might be missing what Zaun might be saying. I don't think they're saying the player asks Monika to impersonate her past self. I think they're saying the player is being silly and asking Monika just to impersonate herself, and Monika takes it in a different direction than the player might expect, sort of in a "careful what you wish for" sort of way.

Yes antimockingjay you get it. We know Monika is smart and clever and somewhat spontaneous and it would be apt imho if this were the "impression" she could give! <3

**Comment by ZaunAura** [comment]
> What I'm trying to say is that I think it would make Monika uncomfortable for the player to ask her to do an impersonation of her past self. I'd imagine it's like thinking about dumb things you used to do and wouldn't want to live through again, and I don't think this topic would exactly fit in.
> 
> You're thinking of using glitches and jumpscares for the topic, but I think we should try to avoid using external elements like those in regular topics, especially jumpscares.
> 
> For the jumpscares, I think using Monika's jumpscare would be insensitive to the player since a lot of people don't like jumpscares. I really don't think that would be a good joke for her to make. For Sayori's hanging jumpscare, willingly going through that again goes against any regrets Monika had about the whole thing. Like I said, I think she would want to avoid going through those events again, especially if the player cares about the other dokis.

Ya the player isn't directly asking her for the impersonation of her past self, it's more of a creative and witty response Monika would give. I don't necesarily think that a jump-scare is necesary but maybe to just stop the music and a cut to black screen fade-in transition to when you first meet Monika in the space classroom is good enough. And even if you let her say a few old lines that she said before MAS takes place then it would definitely not going overboard for any players of MAS to do that.

**Comment by ZaunAura** [comment]
Also, while playing MAS when you ask Monika about if she plays video games, she will say "there's nothing really dangerous here... except for me i suppose." She says this while giving the player a sinister look. That's what I wanted to explore with this suggestion I guess. The player knows Monika can be dangerous and Monika knows she capable of it so I think this "impression" would be the most apt time for such a thing.

**Comment by kimmieclue** [comment]
Okay... Sorry for jumping in quickly but one... Or two.. Things kinda irritated me a bit....
People don’t like jumpscares... Okay... Then... Doki Doki is probably not the game for them in the first place.. For example... I’m not super into jumpscares but I don’t dislike them.. Hence I like the game because if I would not like jumpscares at all I would never become a fan of the game and check out mods
PLUS..
The are jumpscares in the mod..? Like.. If you ask Monika to tell you a horror story... Jumpscares happen... At least once! So it would be the same here (if jumpscares even were implied which they weren’t) that.. Like.. What else do you expect if you click that? Just like with the horror stories, asking Monika to do an impression of herself should kinda show that ‘hey.. might be some scares here so if you don’t want, don’t click’ since it just makes sense!

Sorry for jumping in but I felt I had to because that kinda... Just... It felt hypocritical! So yeah... Heheh! And sorry for bad English, I’m usually fluent but right now I’m super tired heheh!^^

**Comment by Rai99** [comment]
Mainly, I just think that the jumpscare be avoided and any extra special effects should at least be used sparingly. The writing and character would also be an important part, but that'll be looked at if the time comes. 

We'll see what other people and devs think of this idea and see if there are any other concerns or improvements. 

If the topic gets written and put in a PR, we'll see and take it from there. The good/bad news, you can probably expect I'll be there if it happens.

**Comment by kimmieclue** [comment]
Yeah I get that it shouldn’t be used as much since the mod basically is a ‘Monika Girlfriend Simulator’ and all.. But it would be fun with a few small moments where stuff like that could happen considering it would make sense... Of course they don’t have to be many but maybe a few things would be nice. Of course only for things you can choose on your own like the horror stories for example... That way people who don’t want it don’t have to see it. Could even write a small warning like Monika saying it might get glitchy or it might get dark and then ask ‘are you sure you want to see it?’ ^^
Btw.. Sorry for errors and stuff! I’m kind of stressed out so I’m writing this as quick as I can without having the time to check it over^^

**Comment by ZaunAura** [comment]
> Mainly, I just think that the jumpscare be avoided and any extra special effects should at least be used sparingly. The writing and character would also be an important part, but that'll be looked at if the time comes.
> 
> We'll see what other people and devs think of this idea and see if there are any other concerns or improvements.
> 
> If the topic gets written and put in a PR, we'll see and take it from there. The good/bad news, you can probably expect I'll be there if it happens.

ok Rai how do I become a dev? I would like to have some hands on dev experience with MAS and this impression would be good for me since I am motivated for it.

**Comment by Rai99** [comment]
How to become a dev? Well, it's not up to me, but first things first, you're probably not going to become a dev if you're new (ie. you joined git 3 days ago). You'd probably have to show and prove that you have good intentions and can do some things. 

Even then, there's no guarantee you'll be a dev. I think devs are hand picked and recruited from any possible candidate. I think a couple of the most recent devs got their place by being helpful on MASC with questions and tech support. 

Basically, unless you've existed on MASC for a while and been actively helpful, becoming a dev is pretty much out of the question. Try again in a couple months would be my advice, bonus points if we recognize you without having to check.
